api/http/indigo_service/jsonapi.py

```diff
@@ -0,0 +1,570 @@
+#
+# Copyright (C) from 2009 to Present EPAM Systems.
+#
+# This file is part of Indigo toolkit.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import functools
+from enum import Enum
+from typing import Dict, Generic, List, Optional, TypeVar, Union
+
+from pydantic import BaseModel, conlist, validator
+from pydantic.generics import GenericModel
+
+# Generic Types
+
+DataT = TypeVar("DataT")
+ErrorsT = TypeVar("ErrorsT")
+TypeT = TypeVar("TypeT")
+AttributesT = TypeVar("AttributesT")
+OutputFormatT = TypeVar("OutputFormatT")
+
+
+# Formats
+
+
+class CompoundFormat(str, Enum):
+    AUTO = "auto"
+    MOLFILE = "molfile"
+    SMILES = "smiles"
+    CML = "cml"
+    SMARTS = "smarts"
```

Comment:
We can also support Ket format to let Ketcher team try new service.

Reply:
And `InCHi` as well.

Reply:
InCHi added

Reply:
Added KET format. Unfortunately current CompondModel implementation supports structure only as string objects, not dict. It should be fixed in next releases.